### PR TITLE
Doc improvements

### PR DESCRIPTION
**Pull request by pnorman**
* Reference Kosmtik
* Rework yaml->mml
* Fix some caps

### CODE REVIEW

**Comment by matkoniecz** [comment]
After this change "Current work" section will be about past ("The v2.x series initially focused") without mention what is currently done. Section about future mentions reviewing bug reports as a current task what is only small part of what is currently done.

> Avoid adding an overwhelming number of screenshots.

I will try to avoid doing this again :)

**Comment by kocio-pl** [comment]
I'm not sure this remark is needed: nobody else is doing it and Mateusz had a perfect reasons to do it, so it's a just-in-case rule probably, which should be avoided to not make documentation too detailed.

**Comment by math1985** [comment]
> Avoid adding an overwhelming number of screenshots.

+1 for leaving this out. I'd prefer too many screenshots over too little.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,15 +10,45 @@ You should fork the project into your own repo, create a topic branch
...
+`./scripts/yaml2mml.py < project.yaml > project.mml && touch project.mml` to
+update the file and force TileMill to reload it. You shouldn't use the text editor
+built-in to TileMill, it doesn't work with the number of .mss files in the style.
+Instead, hide the right pane and use an external text editor.
+
+Before committing changes, run `./scripts/yaml2mml.py < project.yaml > project.mml`
+to update the JSON MML file and `git add project.mml`
+
+When committing changes, add both the `project.yaml` and `project.mml` files to
+the commit with ./scripts/yaml2mml.py < project.yaml > project.mml && git add project.mml`.
```
Comment (math1985) [inline CONTRIBUTING.md]: Aren't these two lines duplicate with the previous two lines?

File: README.md [hunk 2]
```diff
@@ -45,23 +47,18 @@ no interest in creating further point releases in the v1.x series.
...
-
-There are over [300 open requests][issues], some that have been open for years. These need
-reviewing and dividing into obvious fixes, or additional new features that need some cartographic
-judgement. The work done already in v1.0 and v2.0 will make it much easier to process these.
+There are over [300 open requests][issues], some that have been open for years.
+These need reviewing and dividing into obvious fixes, or additional new features
+that need some cartographic judgement.
 
 [issues]: https://github.com/gravitystorm/openstreetmap-carto/issues
 [cleverness]: https://github.com/openstreetmap/mapnik-stylesheets/blob/master/inc/settings.xml.inc.template#L16
```
Comment (math1985) [inline README.md]: It's not clear to me what this refers to?

**Comment by math1985** [comment]
@pnorman Thanks, very useful!

File: README.md (hunk 2 above)
Comment (pnorman) [inline README.md]: These are existing, and are for the links above.

Comment (math1985) [inline README.md]: But why the link to settings.xml.inc.template (not changed within the current diff actually)?

**Pull request merged**